core/src/main/java/io/questdb/cairo/TableUtils.java

```diff
@@ -432,33 +476,51 @@ public static LPSZ iFile(Path path, CharSequence columnName, long columnTxn) {
         return path.$();
     }
 
-    public static boolean isValidColumnName(CharSequence seq) {
-        for (int i = 0, l = seq.length(); i < l; i++) {
-            char c = seq.charAt(i);
+    public static boolean isValidTableName(CharSequence tableName) {
+        int l = tableName.length();
+        for (int i = 0; i < l; i++) {
+            char c = tableName.charAt(i);
             switch (c) {
-                case '?':
                 case '.':
+                    if (i == 0 || i == l - 1 || tableName.charAt(i - 1) == '.') {
```

Comment:
would be good to add some explanation for these rules. For example, I didn't know Windows does not allow a filename ending with dot.